PyRoute/DeltaDictionary.py

```diff
@@ -0,0 +1,368 @@
+"""
+Created on May 21, 2023
+
+@author: CyberiaResurrection
+"""
+import codecs
+import copy
+import logging
+import os
+import string
+from pathlib import Path
+
+from PyRoute.AllyGen import AllyGen
+from PyRoute.Galaxy import Sector, Allegiance
+from PyRoute.Star import Star
+
+
+class DeltaDictionary(dict):
+
+    def update(self, __m, **kwargs):
+        for key in __m:
+            assert isinstance(__m[key], SectorDictionary), "Values must be SectorDictionary objects"
+        super().update(__m, **kwargs)
+
+    def __setitem__(self, item, value):
+        assert isinstance(value, SectorDictionary), "Values must be SectorDictionary objects"
+        super().__setitem__(item, value)
+
+    def sector_subset(self, sectors):
+        overlap = list()
+        for sector_name in sectors:
+            if sector_name in self:
+                overlap.append(sector_name)
+
+        new_dict = DeltaDictionary()
```

Comment:
Rather than have to deal with the hassles of modifying a DeltaDictionary object, I chose to clone an existing  such object and make whatever changes needed in the process of cloning.

I followed this approach for the component SectorDictionary and SubsectorDictionary objects as well